pkg/tiller/resource_policy.go

```diff
@@ -50,10 +49,12 @@ func filterManifestsToKeep(manifests []Manifest) ([]Manifest, []Manifest) {
 		}
 
 		resourcePolicyType = strings.ToLower(strings.TrimSpace(resourcePolicyType))
-		if resourcePolicyType == keepPolicy {
-			keep = append(keep, m)
+		if resourcePolicyType == deletePolicy {
+			remaining = append(remaining, m)
+			continue
 		}
 
+		keep = append(keep, m)
```

Comment:
The manifest is now appended to the `keep` slice any time the resource policy annotation has been set (and its value is not a variant of `delete`).  This differs from the previous behavior, where `keep` was only updated when `resourcePolicyType == keepPolicy`.

It seems like the end behavior is the same, but I'm not sure if this could cause issues.